modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/GadgetUsageTest.java

```diff
@@ -0,0 +1,252 @@
+                "Pages with the problem is not shown to the user");
+    }
+
+    /**
+     * To test the whether the warnings are updated when a missing gadget is uploaded
+     *
+     * @throws MalformedURLException
+     * @throws XPathExpressionException
+     * @throws InterruptedException
+     */
+    @Test(groups = "wso2.ds.dashboard", description = "Testing the warnings after uploading some of the deleted gadgets",
+            dependsOnMethods = "testWarningInDashboardsAfterDeletingGadgets")
+    public void testWarningAfterUploadingAGadget() throws MalformedURLException, XPathExpressionException,
+            InterruptedException {
+        String systemResourceLocation = FrameworkPathUtil.getSystemResourceLocation();
+        String gadgetFilePath = systemResourceLocation + "files" + File.separator + "publisher.zip";
+        getDriver().get(getBaseUrl() + "/portal/gadget/");
+        getDriver().findElement(By.cssSelector("a[href*='upload-gadget']")).click();
+
+        // Select the correct gadget zip file and upload
+        WebElement inputElement = getDriver().findElement(By.id("selected-file"));
+        getDriver()
+                .executeScript("document.getElementById('selected-file').style.display='block';");
+        inputElement.sendKeys(gadgetFilePath);
+        getDriver().findElement(By.xpath("(//button[@type='button'])[4]")).click();
+        Thread.sleep(1000);
+        redirectToLocation(DS_HOME_CONTEXT, DS_DASHBOARDS_CONTEXT);
+        assertTrue(getDriver().isElementPresent(By.cssSelector("#" + DASHBOARD1_TITLE + " .fw-alert")),
+                "Danger symbol " + "is not displayed in the dashboard when some gadgets are deleted from dashboard");
+        assertFalse(getDriver().isElementPresent(By.cssSelector("#" + DASHBOARD2_TITLE + " .fw-alert")),
+                "Danger symbol is displayed in the " + "dashboard when dashboard has all the gadgets");
+    }
+
+    /**
+     * To create some gadgets in 2 dashboards
+     *
+     * @throws MalformedURLException
+     * @throws XPathExpressionException
+     * @throws InterruptedException
+     */
+    private void createRelevantGadgets() throws MalformedURLException, XPathExpressionException, InterruptedException {
+        redirectToLocation(DS_HOME_CONTEXT, DS_DASHBOARDS_CONTEXT);
+        getDriver().findElement(By.cssSelector("#" + DASHBOARD1_TITLE + " a.ues-edit")).click();
+        String[][] gadgetMappings = { { "publisher", "b" }, { "usa-map", "c" } };
+        String script = generateAddGadgetScript(gadgetMappings);
+        getDriver().navigate().refresh();
+        selectPane("gadgets");
+        Thread.sleep(2000);
+        getDriver().executeScript(script);
+        Thread.sleep(2000);
+        redirectToLocation(DS_HOME_CONTEXT, DS_DASHBOARDS_CONTEXT);
+        getDriver().findElement(By.cssSelector("#" + DASHBOARD2_TITLE + " a.ues-edit")).click();
+        String[][] gadgetMapping = { { "publisher", "b" }, { "subscriber", "d" } };
```

Comment:
This was formatted using wso2 code formatting template for eclipse. So I hope this ok.